Support PT_OPENBSD_RANDOMIZE, PT_OPENBSD_WXNEEDED and PT_OPENBSD_NOBTCFI.

Index: src/passes.cc
--- src/passes.cc.orig
+++ src/passes.cc
@@ -132,6 +132,8 @@ void create_synthetic_sections(Context<E> &ctx) {
 
     if (ctx.arg.z_sectionheader)
       ctx.shdr = push(new OutputShdr<E>);
+
+    ctx.openbsd_randomize = push(new OutputPhdr<E>(PT_OPENBSD_RANDOMIZE));
   }
 
   ctx.got = push(new GotSection<E>);
@@ -196,6 +198,12 @@ void create_synthetic_sections(Context<E> &ctx) {
   if constexpr (is_x86<E>)
     ctx.extra.note_property = push(new NotePropertySection<E>);
 
+  if (ctx.arg.z_wxneeded)
+    ctx.openbsd_wxneeded = push(new OutputPhdr<E>(PT_OPENBSD_WXNEEDED));
+
+  if (ctx.arg.z_nobtcfi)
+    ctx.openbsd_nobtcfi = push(new OutputPhdr<E>(PT_OPENBSD_NOBTCFI));
+ 
   if constexpr (is_riscv<E>)
     ctx.extra.riscv_attributes = push(new RiscvAttributesSection<E>);
 
